source/val/validate_memory.cpp

```diff
@@ -1136,6 +1192,144 @@ spv_result_t ValidateArrayLength(ValidationState_t& state,
   return SPV_SUCCESS;
 }
 
+spv_result_t ValidateCooperativeMatrixLengthNV(ValidationState_t& state,
```

Comment:
Compiler says this function is unused.  Indicates a missing check and missing test coverage.

Reply:
I've hooked it up and added three tests (bad result type, bad operand type, good). I've also (hopefully) fixed the warnings.